### PR TITLE
Support custom agents

### PR DESCRIPTION
**Pull request by brunoocasali**
Related to https://github.com/meilisearch/integration-guides/issues/150 take #2

We will use this door to send information from the Rails gem.

### CODE REVIEW

**Review by alallema — APPROVED** [review]
LGTM! 🌯

File: spec/meilisearch/index/base_spec.rb [hunk 1]
```diff
@@ -126,6 +126,34 @@
...
+      'User-Agent' => "#{custom_agent};#{MeiliSearch.qualified_version}"
+    }
+
+    new_client = MeiliSearch::Client.new(URL, MASTER_KEY, options)
+    new_client.create_index!('options')
+    index = new_client.fetch_index('options')
+    expect(index.options).to eq(options.merge({ convert_body?: true }))
+
+    expect(MeiliSearch::Index).to receive(:get).with(
+      "#{URL}/indexes/options",
```
Comment (bidoubiwa) [inline spec/meilisearch/index/base_spec.rb:143]: A bit confusing to call the index `options`, it looks like a route name and not a resource `indexes/options`. 
But a part for that the test is good!

Comment (brunoocasali) [reply spec/meilisearch/index/base_spec.rb:143]: Agreed, I'll generate an issue to handle that plus a way to randomize the indexes names as we do in the rails :)

**Comment by brunoocasali** [comment]
bors merge

File: spec/meilisearch/index/base_spec.rb (hunk 1 above)
Comment (brunoocasali) [reply spec/meilisearch/index/base_spec.rb:143]: Issue created! #355

**Pull request merged**